### ISSUE TITLE
Codegen is not using fragments on version 0.7.0-alpha.5

### ISSUE THREAD

**Issue by aryan-ghassemi-ck** [apollographql/apollo-ios#155]
There seem to be a bug in version `0.7.0-alpha.5` where the generated code is not using fragments, instead it's inlining fields

```
fragment formattedTextFragment on FormattedText {
	span {
		text
	}
}
```
```
// Content is a union
content {
    ... on FormattedText {
        ...formattedTextFragment
    }
    ... on FormattedList {
    	items {
    		...formattedTextFragment
    	}
    }
}
```
In the previous codegen, this would result in a single `Span` struct under `formattedTextFragment`. In the new Codegen we have 3 separate structs for `Span`

The class Structure went:
from
```
Content -> AsFormattedList -> Item -> Fragments -> FormattedTextFragment -> Span
Content -> AsFormattedText -> Fragments -> FormattedTextFragment

```
to
```
Content -> AsFormattedList -> Item -> Span
Content -> AsFormattedText -> Span
```
This breaks all the extensions we have written on top of our fragment structs, because it's no longer reusing them. 

NOTE:
We run the codegen with `merge-in-fields-from-fragment-spreads` set to `false`

**Comment by martijnwalraven** [comment]
This should be fixed in Apollo iOS 0.7.0-alpha.6. Let me know how it works for you!

**Comment by aryan-ghassemi-ck** [comment]
That solved the issue we were seeing, thanks
